src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -330,10 +369,23 @@ internal static Type ResolveTypeNameWithContext(TypeName typeName, out Exception
                         assemList = ClrFacade.GetAssemblies(typeResolutionState, newTypeName);
                     }
 #endif
-                    var newResult = ResolveTypeNameWorker(newTypeName, currentScope, typeResolutionState.assemblies, typeResolutionState, /* reportAmbiguousException */ true, out exception);
+                    var newResult = ResolveTypeNameWorker(newTypeName, currentScope, typeResolutionState.assemblies, searchedAssemblies, typeResolutionState,
+                                                          /*onlySearchInGivenAssemblies*/ false, /* reportAmbiguousException */ true, out exception);
                     if (exception == null && newResult == null)
                     {
-                        newResult = ResolveTypeNameWorker(newTypeName, currentScope, assemList, typeResolutionState, /* reportAmbiguousException */ false, out exception);
+                        if (context != null && assemblies == null)
```

Comment:
`assemblies` is the assemblies explicitly passed into this method to search from. If the caller does specify the assemblies, then we don't want to search AssemblyCache.